webapp/src/plugin.jsx

```diff
@@ -52,16 +52,20 @@ export default class DemoPlugin {
         registry.registerBottomTeamSidebarComponent(
             BottomTeamSidebar,
         );
-        const {showRHSPlugin} = registry.registerRightHandSidebarComponent(
+        const {showRHSPlugin, hideRHSPlugin} = registry.registerRightHandSidebarComponent(
             RHSView,
             <FormattedMessage
                 id='plugin.name'
                 defaultMessage='Demo Plugin'
             />);
 
+        let actionButtonIsActive = false;
         registry.registerChannelHeaderButtonAction(
             <ChannelHeaderButtonIcon/>,
-            () => store.dispatch(showRHSPlugin),
+            () => {
+                store.dispatch(actionButtonIsActive ? hideRHSPlugin : showRHSPlugin);
```

Comment:
If I understand correctly, if I execute the following steps:
* click channel header button
* click flagged posts icon
* click channel header button

the RHS will be closed instead of replacing the flagged posts RHS with the demo plugin again. This might be unintuitive. @marianunez, is it possible to know if your RHS component is currently active?

Reply:
Based on @Lisenish changes here https://github.com/mattermost/mattermost-webapp/pull/3522, dispatching `hideRHSPlugin` will have no effect if the RHS is not showing your plugin. So, in your example it would not close the RHS but I agree it would not show the Demo Plugin RHS again. 

We currently don't have anything explicit that let's plugins know if there is anything open unless they check directly in the store which does know the state of the RHS.

Reply:
@lieut-data @marianunez Yeah I don't like it too, actually I have comment about it but forgot to post me review yesterday, please take a look below.

And yes currently we have somewhat buggy behaviour with this plugin because of that implementation, like @marianunez mentioned if we click on the button in your case it will do nothing but I think it's expected to open RHS plugin panel again, but we should click 2nd time for this to happen now.